tests/loggen/loggen_plugin.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2007-2018 Balabit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef __LOGGEN_PLUGIN_H__
+#define __LOGGEN_PLUGIN_H__
+
+#include <glib.h>
+#include <gmodule.h>
+
+typedef struct _plugin_option
+{
+  int message_length;
+  int interval;
+  int number_of_messages;
+  int permanent;
+  int active_connections;
+  int idle_connections;
+  int use_ipv6;
+  const char *target; /* command line argument */
+  const char *port;
+  int  rate;
+} PluginOption;
+
+typedef struct _thread_data
+{
+  PluginOption *option;
+  int index;
+  int sent_messages;
+  struct timeval start_time;
+  struct timeval last_throttle_check;
+  long buckets;
+} ThreadData;
```

Comment:
It is true that ThreadData is for internal use of plugins only. As we discussed personally the I want to keep it in a common header file to avoid the big number of headers.